app/scripts/services/DataRepository.js

```diff
@@ -289,6 +315,42 @@ var DataRepository = function (dataContainer, syncObject, recordSets) {
     return id;
   };
 
+  this.determineIndexFromBrandModel = function determineIndexFromBrandModel(brandModel) {
+    // Determine the index of the firebase array, using the brand model's id.
+    var indexToDelete = null;
```

Comment:
Are there any other places we may want to use this function? If there are we may want to rename this variable so it isn't related to deletion. Maybe something like `targetIndex` or `indexMatch` or something else completely :) . (same in the other 2 functions below)

Reply:
Ah yes, you're right. Well spotted! I re-used the code that was explicitly for deleting a record, but didn't notice the variable name.
I'll change it now - this method is also used when updating a record.

Reply:
I'm afraid I went with a different format for the name :-)
I went with brandIndex, categoryIndex, etc.
The [commit is here](https://github.com/elixirlabsinc/endslaverynow/commit/b0acd5b5ce9bb4b8f56b5f9152df8b4667504ae6).

Reply:
Perfect - love it!